src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -259,7 +259,7 @@ internal void RequestWorker()
         {
             Interlocked.Increment(ref _numRequestedWorkers);
             WorkerThread.MaybeAddWorkingWorker();
-            // TODO: Ensure gate thread is running here.
+            GateThread.EnsureRunning();
```

Comment:
Would you be commenting this out before replacing the existing thread pool implementation? I think it would be good to test the thread pool implementation without the gate thread or hill climbing to reduce the testing scope and stage the changes, increasing the confidence of a smaller component when it is enabled. It would be useful to get something working end-to-end before adding other features of the thread pool. Then each feature can be tested and added independently to an already working implementation.